### PR TITLE
RND-168 Add dell icons css file

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: https://cloudifysource.atlassian.net/browse/RND-168

### Other PRs
https://github.com/cloudify-cosmo/cloudify-stage/pull/2494

## Screenshots / Videos
Screenshot 2023-03-27 at 5 37 48

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
N/A

 

## Documentation
Will be done at the same time of finalising the PR in `cloudify-tage`.

### CODE REVIEW

**Comment by Vorbert-Kruk** [comment]
As @leon-good-life you may have some agreements with @qooban, when it comes to the implementation of dell's icons, I'd like to wait till @qooban would review this PR
I may have some suggestions already, but I'd want to make sure that they won't be commented unnecessarily

File: packages/frontend/styles/dds-icons.css
Comment (Vorbert-Kruk) [inline packages/frontend/styles/dds-icons.css:27]: We can deduplicate `dds__icon` selector in a simple way:

```suggestion
@font-face {
  font-family: "dds-icons";
  font-style: normal;
  font-weight: normal;
  src: url("//dds.dell.com/icons/2.8.0/dds-icons.woff2") format("woff2"), url("//dds.dell.com/icons/2.8.0/dds-icons.woff") format("woff");
}
.dds__icon {
  font-family: "dds-icons" !important;
  fill: currentColor;
  width: 1em;
  height: 1em;
  -webkit-font-smoothing: auto;
  -moz-osx-font-smoothing: auto;
  font-size: inherit;
  font-style: normal;
  font-feature-settings: normal;
  font-variant: normal;
  font-weight: normal;
  text-transform: none;
  line-height: 1em;
}
```

File: packages/frontend/styles/dds-icons.css
Comment (Vorbert-Kruk) [inline packages/frontend/styles/dds-icons.css:2]: @qooban are we okay with that naming convention, or should we try to adjust it? 🤔 
All selectors in this file are written in [BEM naming convention](https://en.bem.info/methodology/naming-convention/) which is a good thing, but it's not consistent with the way of operating on class names on our project

Because of that I wonder if we should adjust it or leave it as it is 🤔 
I know that we could e.g. make a dictionary, which we could use to simplify usage of those icons and to omit confusion about the naming convention, but still I'm wondering which direction seems to be optimal
WDYT? 

Maybe this questions should be addressed during the implementation, not while introducing this file to repository 🤔

Comment (leon-good-life) [reply packages/frontend/styles/dds-icons.css:2]: I copied this file from Dell Design Library (https://www.delldesignsystem.com/foundations/icons/). I can't find the repository to check how often it is updated if at all. But I think if it is being updated we should consider that we may need to update this file in the future if Dell updates it. Looking what lines were changed and refactor it every time may be time consuming. So maybe making an exception for being inconsistent here is worth it?

Comment (Vorbert-Kruk) [reply packages/frontend/styles/dds-icons.css:2]: Yeah, you're mentioning a valid point
As I'm also wondering about it, I'd wait for the @qooban voice on that subject

### Note
After deeper thought, I wouldn't be worried about updating stylesheet related to icons, as from current point of view we wouldn't need more icons that are presented in the stylesheet and there is no indication how often icons library is being modified (I've tried to get that information from [changelog page](https://www.delldesignsystem.com/updates/changelog/))
Looking from other libraries perspective, icons are one of the library parts which is not often updated, once it's being included into a specific library
So IMO we shouldn't be worried about the timeliness aspect 💪

Comment (qooban) [reply packages/frontend/styles/dds-icons.css:2]: 1. We will be moving towards Dell Design System AFAIK, but it's uncertain how we will do the switch and which parts will be affected, which parts will have to be rewritten. We haven't finished switch to new theme fully yet and another one is coming in, it adds to complexity even more 🙂 
2. RND-169 won't solve all our concerns about DDS switch as it's big topic of course, it should be treated only as first, small step.
3. If we wouldn't have that offline environment requirement (mentioned in [this comment](https://cloudifysource.atlassian.net/browse/RND-168?focusedCommentId=79534)), we would just import the file directly as is and not store it. Maybe that requirement will be dropped in new Dell world, who knows ;-) 

... so in this specific case I think we shouldn't invest more time in adjustments, I'm good with having this CSS as is. To address inconsistency concern, maybe we could have a simple function for getting DDS icons class names, so that these classes will be used directly only in one place? 🤔 To be placed either in this repo or in cloudify-stage (for the time being). It would also be reasonable to have a new type (if DDS is not already providing something) to narrow down icon-name argument type, not to have any string, but predefined list of strings.

Anyways, I'm adding another point to my discussions with Alex B. and Alex M. - how we will do switch to DDS?

Comment (Vorbert-Kruk) [reply packages/frontend/styles/dds-icons.css:2]: Sure, in that case let's leave it as it is 💪 
I'm resolving all discussions as they're requiring some modifications from the code perspective

**Review by Vorbert-Kruk — APPROVED** [review]

**Comment by qooban** [comment]
@leon-good-life I talked with Alex M. today and I'd like to say: please don't worry about RND-168, it's not critical item, it's not planned for 7.0. Put work aside and spend time with your girlfriend and enjoy your vacation! 🙂

File: packages/frontend/styles/dds-icons.css
Comment (leon-good-life) [reply packages/frontend/styles/dds-icons.css]: [aad209c](https://github.com/cloudify-cosmo/cloudify-ui-common/pull/157/commits/aad209ca25eb2a6dd9750bee5e5a0f0490f41eac)

**Review by qooban — APPROVED** [review]
I'm still not sure how we will do the switch to DDS, but... not to block you I'm approving this PR.

**Comment by qooban** [comment]
@leon-good-life Before merging and releasing new minor version, please make sure it works properly with Stage.

**Pull request merged**
